src/main/java/com/google/sps/DataServlet.java

```diff
@@ -164,37 +200,57 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       return;
     }
 
-    List<String> queried = new ArrayList<>();
-    if (nodeNameParam.length() > 0) {
-      queried.add(nodeNameParam);
-    }
-    // If the token is contained, then get the nodes associated with the token and
-    // add them to the queried nodes
+    // Show mutations relevant to nodes that contain the token in the new graph
     if (currDataGraph.tokenMap().containsKey(tokenParam)) {
       queried.addAll(currDataGraph.tokenMap().get(tokenParam));
+      queriedNext.addAll(currDataGraph.tokenMap().get(tokenParam));
     }
+
+    // Truncate the graph from the nodes that the client had searched for
     truncatedGraph = currDataGraph.getReachableNodes(queried, depthNumber);
 
-    if (nodeNameParam.length() == 0 && truncatedGraph.equals(currDataGraph.graph())) {
-      // If we are not filtering the graph or limiting its depth, show all mutations of all nodes
+    // The nodes to calculate relevant mutations from
```

Comment:
Since this is a `truncatedGraph`, this comment is a bit inaccurate since it seems to indicate "a list of nodes" :)